addons/beehave/nodes/beehave_root.gd

```diff
@@ -47,12 +61,17 @@ func get_last_condition_status():
 			return "RUNNING"
 	return ""
 
-
 func enable():
 	self.enabled = true
-	set_physics_process(true)
-
+	set_process(process_mode == PROCESS_MODE.IDLE)
+	set_physics_process(process_mode == PROCESS_MODE.PHYSICS_PROCESS)
 
 func disable():
 	self.enabled = false
-	set_physics_process(false)
+	set_process(self.enabled)
+	set_physics_process(self.enabled)
+
+func set_process_mode(value):
+	process_mode = value
+	set_process(process_mode == PROCESS_MODE.IDLE)
```

Comment:
A bit of a funny one: if you enable the idle thread from within the physics thread there might be a race condition where it starts processing the tree already while this very method is still running. Be mindful of that edge case.

Reply:
Good catch! Should I add a note, or make it wait until the next tick to change the process mode? Might be able to do the latter with the signals from the debug PR

Reply:
For now, add a FIXME comment. Let's see if someone has issues in future and we might implement a fix for it.